core/finality-grandpa/src/lib.rs

```diff
@@ -688,12 +688,26 @@ where
 	}
 }
 
+impl<B, E, Block: BlockT<Hash=H256>, RA, PRA> ProvideRuntimeApi for GrandpaBlockImport<B, E, Block, RA, PRA>
```

Comment:
Why do you need to implement this here? As far as I see, you don't use it?

Reply:
It is used in `node-service`